### PR TITLE
Bug 1538943: Image carry out the deployment should be optional for Custom deployment strategy and should have unique ID

### PR DESCRIPTION
**Pull request by jhadvig**
For Custom deployment strategy, the Image Name which will carry out the deployment should be optional per docs at https://docs.openshift.com/container-platform/3.7/dev_guide/deployments/deployment_strategies.html#custom-strategy

As part of this PR I also updated the `id`'s in the template to match `id's` of the `help-block`
@spadgett PTAL

### CODE REVIEW

File: app/views/edit/deployment-config.html [hunk 1]
```diff
@@ -92,7 +91,7 @@ <h3>Deployment Strategy</h3>
                               aria-describedby="strategyTimeout">
```
Comment (spadgett) [inline app/views/edit/deployment-config.html]:
```html
aria-describedby="strategy-timeout-help"
```

File: app/views/edit/deployment-config.html [hunk 2]
```diff
@@ -127,7 +126,7 @@ <h3>Deployment Strategy</h3>
                                   aria-describedby="updatePeriod">
```
Comment (spadgett) [inline app/views/edit/deployment-config.html]:
```html
aria-describedby="update-period-help"
```

File: app/views/edit/deployment-config.html [hunk 3]
```diff
@@ -159,7 +158,7 @@ <h3>Deployment Strategy</h3>
                                   aria-describedby="interval">
```
Comment (spadgett) [inline app/views/edit/deployment-config.html]:
```html
aria-describedby="interval-help"
```

File: app/views/edit/deployment-config.html [hunk 4]
```diff
@@ -212,7 +211,7 @@ <h3>Deployment Strategy</h3>
                                 class="form-control"
                                 aria-describedby="maxSurge">
```
Comment (spadgett) [inline app/views/edit/deployment-config.html]: max-surge-help

**Comment by jhadvig** [comment]
@spadgett comments addressed. PTAL

**Pull request merged**